src/antsibull/sanity_tests.py

```diff
@@ -0,0 +1,361 @@
+# Copyright (C) 2023 Maxwell G <maxwell@gtmx.me>
+# SPDX-License-Identifier: GPL-3.0-or-later
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+"""
+Utilities to run sanity tests accross collections
+"""
+
+from __future__ import annotations
+
+import datetime
+import json
+import shlex
+import shutil
+import sys
+from collections.abc import Collection, Iterable, Iterator, Sequence
+from dataclasses import dataclass
+from functools import partial
+from pathlib import Path
+from subprocess import CompletedProcess
+from typing import TYPE_CHECKING, Any, TypedDict
+
+from antsibull_core import app_context
+from antsibull_core.logging import log
+from antsibull_core.subprocess_util import log_run
+from antsibull_core.yaml import store_yaml_file
+from packaging.version import Version
+
+from antsibull.constants import SANITY_TESTS_BANNED_IGNORES, SANITY_TESTS_DEFAULT
+from antsibull.types import CollectionName, add_dataclass_yaml_type
+
+if TYPE_CHECKING:
+    from _typeshed import StrPath
+
+mlog = log.fields(mod=__name__)
+
+
+class SanityOutput(TypedDict):
+    """
+    Mapping of `ansible-test sanity` output and other related data
+    """
+
+    cmd: list[str]
+    returncode: int
+    stdout: str
+    stderr: str
+    runtime: float
+    test_json: dict[str, Any]
+    ignore_entries: list[IgnoreEntry]
+    banned_ignore_entries: list[IgnoreEntry]
+    ignores_file: str | None
+
+
+class CollectionOutput(TypedDict):
+    """
+    Collection entry
+    """
+
+    failed: bool
+    sanity: SanityOutput
+
+
+class EnvDetails(TypedDict):
+    """
+    Details about the ansible-test environment
+    """
+
+    ansible_test_version: str
+    sanity_tests: list[str]
+
+
+class Output(TypedDict):
+    """
+    Parent mapping of the sanity test data file
+    """
+
+    collections: dict[CollectionName, CollectionOutput]
+    total_runtime: float
+    env_details: EnvDetails
+
+
+@dataclass(frozen=True)
+class IgnoreEntry:
+    """
+    Represents an entry in an `ansible-test sanity` ignores file
+    """
+
+    file: str
+    test: str
+    remainder: str
+
+    def as_str(self) -> str:
+        return f"{self.file} {self.test}{self.remainder}"
+
+
+add_dataclass_yaml_type(IgnoreEntry)
+
+
+def parse_ignores_file(file: StrPath) -> Iterator[IgnoreEntry]:
+    """
+    Parse a sanity test ignore file
+    """
+    flog = mlog.fields(func="parse_sanity_test_file", file=file)
+    with open(file, encoding="utf-8") as fp:
+        for line in fp:
+            line = line.rstrip("\n")
+            parsed = line.split(" ", 2)
+            if len(parsed) == 2:
+                parsed.append("")
+            if len(parsed) != 3:
+                flog.error("Failed to parse line: {0}", line)
+                continue
+            yield IgnoreEntry(file=parsed[0], test=parsed[1], remainder=parsed[2])
+
+
+def get_ignores_file(directory: Path, version: str) -> Path | None:
```

Comment:
Shouldn't this be `get_ignores_file`? This is returning a single file (or none), not multiple ones.

Reply:
You're saying it should be `get_ignore_file()`?

Reply:
I don't care whether it's ignore (referencing the filename `ignore-xxx.txt`) or ignores (referencing that the file's content is a list of errors to ignore), but it should be file, not files :)

Reply:
The function is currently named `get_ignores_file()`. There's no plural `files` anywhere in the name.